traitlets/config/configurable.py

```diff
@@ -187,6 +187,10 @@ def _config_changed(self, change):
 
     def update_config(self, config):
         """Update config and load the new values"""
```

Comment:
Given these specific semantics (deepcopy first before update), I think it would be a good idea to extend a little bit the docstring with a note that indicates that's what happens, so it becomes clearly official that such semantics are the intended ones.